libuavcan_drivers/linux/include/uavcan_linux/socketcan.hpp

```diff
@@ -315,6 +328,22 @@ class SocketCanIface : public uavcan::ICanIface
         }
     }
 
+    /**
+     * Returns true if a frame accepted by HW filters
+     */
+    bool checkHWFilters(const uavcan::CanFrame frame)
+    {
+        uint16_t container_size = hw_filters_container_.size();
+        for (uint16_t i = 0; i < container_size; i++)
+        {
+            if (((frame.id & hw_filters_container_[i].mask) ^ hw_filters_container_[i].id) == 0)
+            {
+                return 1;
+            }
+        }
+        return 0;
```

Comment:
You need to use explicit true/false with boolean data type. Implicitly casting int to bool is unsafe, and many code analysis tools will rightfully complain about it.